store/structured/mongo/store.go

```diff
@@ -70,16 +83,30 @@ func (o *Store) Initialize(ctx context.Context) error {
 }
 
 func (o *Store) GetRepository(collection string) *Repository {
+	if o.client == nil {
```

Comment:
Perhaps `panic` is better, because this is not meant to be called before the store is initialized.